plugins/postcss-text-decoration-shorthand/src/index.ts

```diff
@@ -145,9 +145,25 @@ const creator: PluginCreator<pluginOptions> = (opts?: pluginOptions) => {
 
 					const nonShortHandValue = valueParser.stringify(data.line);
 					if (decl.value.toLowerCase() === nonShortHandValue) {
+						const next = decl.next();
+						if (!next || next.type !== 'decl' || next.prop.toLowerCase() !== 'text-decoration') {
```

Comment:
Have to look at `next`, not previous because we only want to do this for declarations written by authors. Not declarations produced by this plugin.